server/cars.js

```diff
@@ -75,13 +117,60 @@ router.get('/:carId', auth.verifyToken, User.verify, async (req, res) => {
     let car = await Car.findOne({
       _id: req.params.carId
     });
+
+    const nextOil = getNextService(car.mileage, intervals.OIL, car.oil);
+    const nextTransmission = getNextService(car.mileage, intervals.TRANSMISSION,car.transmission);
+    const nextTire = getNextService(car.mileage, intervals.TIRE_ROTATION, car.tire);
+    const nextCoolant = getNextService(car.mileage, intervals.COOLANT, car.coolant);
+    const nextAir = getNextService(car.mileage, intervals.AIR_FILTER, car.engine);
+
+    car._doc.maintenanceItems = [
+      {
+        title: 'Oil Change',
+        next: nextOil,
+        percent: Math.round(car.mileage / nextOil),
+        color: '#ec407a'
+      },
+      {
+        title: 'Transmission Fluid',
+        next: nextTransmission,
+        percent: Math.round((car.mileage / nextTransmission) * 100),
+        color: '#f4511e'
+      },
+      {
+        title: 'Tire Rotation',
+        next: nextTire, 
+        percent: Math.round((car.mileage / nextTire) * 100),
+        color: '#512da8'
+      },
+      {
+        title: 'Air Filter',
+        next: nextAir, 
+        percent: Math.round((car.mileage / nextAir) * 100),
+        color: '#f57c00'
+      },
+      {
+        title: 'Coolant Flush',
+        next: nextCoolant, 
+        percent: Math.round((car.mileage / nextCoolant) * 100),
+        color: '#0288d1'
+      }
+    ]
+
+    car._doc.estimatedCurrentMileage = calculateCurrentMileage(car._doc.mileage, car._doc.averageMPY, car._doc.created)
```

Comment:
This might be handy for the display and it gives a purpose to collecting the average miles driven per year